airflow/sensors/sql.py

```diff
@@ -58,21 +61,22 @@ class SqlSensor(BaseSensorOperator):
     ui_color = '#7c7287'
 
     def __init__(
-        self, *, conn_id, sql, parameters=None, success=None, failure=None, fail_on_empty=False, **kwargs
+        self, *, conn_id, sql, parameters=None, success=None, failure=None, fail_on_empty=False, hook_params={}, **kwargs
     ):
         self.conn_id = conn_id
         self.sql = sql
         self.parameters = parameters
         self.success = success
         self.failure = failure
         self.fail_on_empty = fail_on_empty
+        self.hook_params = hook_params
         super().__init__(**kwargs)
 
     def _get_hook(self):
         conn = BaseHook.get_connection(self.conn_id)
 
         allowed_conn_type = {
-            'google_cloud_platform',
+            'gcpbigquery',
```

Comment:
As I understand, to use some connection in this sensor it hook should be an instance of DbApiHook.
But GoogleBaseHook which returns for the google_cloud_platform isn't that one.
Am I correct?

Reply:
Yeah, this certainly isn't a list of all DbApiHooks, since its missing `druid` and `eleasticsearch` amongst others.

Reply:
We actually check for subclass of `DbApiHook` in:
https://github.com/apache/airflow/blob/dd410fd3c9de14e94034dcb4ccae52bbf5216199/airflow/operators/sql.py#L68-L73

So we should do here the same.
I'm not sure why you mention druid and elasticsearch as exceptions?
https://github.com/apache/airflow/blob/dd410fd3c9de14e94034dcb4ccae52bbf5216199/airflow/providers/apache/druid/hooks/druid.py#L131

https://github.com/apache/airflow/blob/dd410fd3c9de14e94034dcb4ccae52bbf5216199/airflow/providers/elasticsearch/hooks/elasticsearch.py#L27

Reply:
Done